### DISCUSSION TITLE
how to save the model in every epochs?

### DISCUSSION THREAD

**Discussion by playezio** [ultralytics/yolov5#7883] [Q&A]
# Save model
            if (not nosave) or (final_epoch and not evolve):  # if save
                ckpt = {
                    'epoch': epoch,
                    'best_fitness': best_fitness,
                    'model': deepcopy(de_parallel(model)).half(),
                    'ema': deepcopy(ema.ema).half(),
                    'updates': ema.updates,
                    'optimizer': optimizer.state_dict(),
                    'wandb_id': loggers.wandb.wandb_run.id if loggers.wandb else None,
                    'date': datetime.now().isoformat()}

                # Save last, best and delete
                torch.save(ckpt, last)
                if best_fitness == fi:
                    torch.save(ckpt, best)
                if (epoch > 0) and (opt.save_period > 0) and (epoch % opt.save_period == 0):
                    torch.save(ckpt, w / f'epoch{epoch}.pt')
                del ckpt
                callbacks.run('on_model_save', last, epoch, final_epoch, best_fitness, fi)

this code can save the last and best model, but cant save the model in every epoch. how to modify that code to save all models?

**Comment by nidetaoge** [comment]
In argparse params, just remove '--nosave' and add '--save_period 1' to save the model in every epoc.

**Reply by glenn-jocher** [reply]
`python train.py --save-period 1` will save every epoch

**Reply by SJavad** [reply]
> `python train.py --save-period 1` will save every epoch

how can I save the offline experiment comet zip file in every epoch?
